src/components/Layout/components/getThemeValue.ts

```diff
@@ -0,0 +1,31 @@
+import {LayoutProps, LayoutTheme, MediaPartial} from '../types';
+
+/**
+ * Select provided prop value from LayoutTheme. Depends on the current screen media query
+ */
+export function getThemeValue<T extends keyof LayoutProps>(
+    key: T,
+    theme: Partial<LayoutTheme>,
```

Comment:
Why layout is connected with theme at all? Theme should only affect colors, not other things.

Reply:
You can use `LayoutProvider` as a separate provider now. Current `ThemeProvider` only responsible for color app theme and should be called `ColorThemeProvider`.

But I think it will be inconvenient for the consumer to connect several providers to his application, because he will have to figure out what each of these providers is responsible for